cli/CMakeLists.txt

```diff
@@ -35,12 +36,14 @@ int main(int argc, char **argv) {
 		endif()
 	endif()
 
+	set(CMAKE_REQUIRED_INCLUDES)
 	set(CMAKE_REQUIRED_LIBRARIES)
 endif()
 
 if (READLINE_FOUND)
 	list(APPEND CLI_SOURCES CommandLine.cpp)
 	list(APPEND CLI_LIBRARIES ${Readline_LIBRARY})
+	include_directories(SYSTEM ${Readline_INCLUDE_DIR})
```

Comment:
please use `target_include_directories` for cli target (I reckon it should be PRIVATE)